### DISCUSSION TITLE
Filter on a Relationship

### DISCUSSION THREAD

**Discussion by AndrewMarkUK** [dedoc/scramble#216] [Q&A]
The json outputs record lookings like this:

        {
            "id": 8,
            "action_name": "View",
            "action_code": "08",
            "action_description": "opened the event",
            "action_group_id": 6,
            "monitoring": null,
            "type": false,
            "actionGroup": {
                "action_group_id": 6,
                "group_name": "VISUALIZE",
                "action_device": false,
                "manual_action_group": true
            }
        }

actionGroup is a relationship. Is it possible to make it so that, in the documentation page we can provide a parameter input for the group_name in that relationship so that when testing we can provide a string (in this case it would be VISUALIZE) and then it would only return matches when we click 'Sending API Request'.

How do we go about making that available? Essentially we want to filter on actionGroup > group_name.

**Comment by romalytvynenko** [comment]
@AndrewMarkUK Hey!

Sorry, I'm not sure I understand what you want to achieve

**Reply by AndrewMarkUK** [reply]
Hello, I will try and clarify.

In /docs/api#/operations/actions.index I wish to include in the Parameters Box where the Send API Button is, a field for the actionGroup.group_name field in the above object. Currently I have a box for id  and action_view but when testing, I would also like to include an input for  actionGroup.group_name < field within the object, but I am not finding a way of doing this. For the main model I am doing something like:

            /** @var string $content varchar(20) */
            'action_name' => Str::title($this->acao_nome),

and that works fine, it shows a input field we can test.

But how to do it for groupAction.group_name which is field inside a relationship?

Hope that's clearer.

**Comment by romalytvynenko** [comment]
@AndrewMarkUK I afraid this is not possible, and I won't be able to add it as it would be a feature of UI (which is handled by other library).

**Comment by AndrewMarkUK** [comment]
Just to be clear, in documentation, When using 'Send Api...' test button, we cannot filter on a relation?

**Reply by romalytvynenko** [reply]
@AndrewMarkUK you can filter but with whatever options you see there, in UI. If there is no option for filtering by some field - then nope.

Andrew, sorry but I have troubles with understanding "filter on a relation".

Maybe share some screenshots and annotate what you expect to have? Because I'm not getting it

**Comment by AndrewMarkUK** [comment]
![Docs_Snapshot](https://github.com/dedoc/scramble/assets/8135582/52080943-9f5c-465c-a834-ce3c39460a26)
![Rules_Validation](https://github.com/dedoc/scramble/assets/8135582/ab5691de-5891-4bb3-bdfa-c6a2ab9f91f3)

Hopefully, this will be more clear. You can see there is input for group_name and that group_name belongs to actioGroup. I have also included the code where the validation. It might be that I am including trh wrong syntax for group_name in the validation rules?

**Reply by romalytvynenko** [reply]
Oh, I see.

Sorry, but this is not about docs generator or docs UI. This is missing implementation of this filter in your codebase.

**Reply by AndrewMarkUK** [reply]
Yes, I thought this was the case, So I ask, how is the filter created as I am trying different ways.
